distsql/distsql.go

```diff
@@ -158,5 +160,42 @@ func canUseChunkRPC(ctx sessionctx.Context) bool {
 	if ctx.GetSessionVars().EnableStreaming {
 		return false
 	}
+	if !checkAlignment() {
+		return false
+	}
 	return true
 }
+
+var supportedAlignment = !(unsafe.Sizeof(types.MysqlTime{}) != 16 ||
+	unsafe.Sizeof(types.Time{}) != 20 ||
+	unsafe.Sizeof(types.MyDecimal{}) != 40)
+
+// checkAlignment checks the alignment in current system environment.
+// The alignment is influenced by system, machine and Golang version.
+// Using this function can guarantee the alignment is we want.
+func checkAlignment() bool {
+	return supportedAlignment
+}
+
+// SetSystemEndian sets the system endian for the DAGRequest.
+func SetSystemEndian(dagReq *tipb.DAGRequest) {
```

Comment:
```suggestion
func setSystemEndian(dagReq *tipb.DAGRequest) {
```

It need not be exported.